### ISSUE TITLE
Max Bounty Hunter Issues - IV

### ISSUE THREAD

**Issue by pavlovcik** [ubiquity/ubiquibot#428]
We should let bounty hunters start another bounty if they have at least one approving review on their pending pull requests.

This should also be configurable (value of `0` to disable this. `1` for one approval etc.)

---

> There are 2 pending PRs ([one](https://github.com/ubiquity/ubiquity-dollar/pull/704) and [two](https://github.com/ubiquity/ubiquity-dollar/pull/702)) but they seem to be acceptable so I guess we can bypass the rules and assign manually

_Originally posted by @rndquu in https://github.com/ubiquity/ubiquity-dollar/issues/695#issuecomment-1604343393_

**Comment by wannacfuture** [comment]
We are currently just checks the PR if it has review.
So instead of this, we check if PR has **APPROVED** review?

**Comment by pavlovcik** [comment]
Yes this is another enhancement. That's why the issue title ends in Roman numerals of 4. It's the fourth enhancement to the idea.

**Comment by wannacfuture** [comment]
lol, This issue must be solved to assign myself to THIS issue in this situation because I have one approved PR🤣.
@pavlovcik , can I assign myself manually?

**Comment by pavlovcik** [comment]
- If everybody on the review team didn't get a chance to leave a review on an open pull request in a full 24 hours, the bounty hunter should be able to continue working on new tasks. 
- If there is any approval on a review, then the bounty hunter should be able to work on another task too.

This means that if there is a "left a comment" or "changes requested" review state, then they are blocked from starting a new bounty. 

Yes this means that its okay if they slip through at hour 24, start a new bounty, and then at hour 25 somebody on the review team marks the pull request as "changes requested" which means that the bounty hunter currently is assigned to 3 tasks but is restricted from starting any more.

**Comment by wannacfuture** [comment]
>If there is any approval on a review from the core team (im pretty sure we can set this in the repository settings, to restrict any community members from doing reviews, for example) then the bounty hunter should be able to work on another task too.


I think only core team can do review any prs(am I right?) so we can just check if there is any approved review(I mean, not needed to check if the reviewer is core team).

**Comment by pavlovcik** [comment]
I think you're right. I just double checked the GitHub settings and looks like it's only possible for core team members to leave a review.
